var/spack/repos/builtin/packages/ibm-databroker/package.py

```diff
@@ -7,7 +7,7 @@
 from spack.package import *
 
 
-class IbmDatabroker(CMakePackage, PythonPackage):
+class IbmDatabroker(CMakePackage, PythonExtension):
```

Comment:
> Originally, inheriting CMakePackage and PythonPackage would include the phases defined in both build systems, but .phases would be inherited from only one. Do you know which build system "won" before?

`CMakePackage`, due to how method resolution order is computed. Would be the opposite if the order of inheritance is inverted.

> I think it would be useful to create a term that applies to inherited packages which are intended to provide useful behavior but are not used for building (e.g. PythonExtension, CudaPackage, etc.).

My thoughts on this:
1. We can rename `PythonExtension` to anything else, this is a new class that appears in the PR. I named it that way because people usually need it if they are packaging a "native python extension", i.e. an extension for Python coded e.g. in C or C++ and built with another build system. If people are using a `PythonPackage` they inherit from `PythonExtension` already.
2. For `CudaPackage` etc. we have to deal with the backward compatibility issue. We can rename the class (though `Essentials` seems a bit generic) and substitute it everywhere in the `builtin` repository. Then we need to decide if we want to expose the old name and deprecate it, or just error out. 
 
If possible I'd avoid dealing with 2 in this PR.

Reply:
> If possible I'd avoid dealing with 2 in this PR.

That sounds fine

Perhaps I'm chasing the wrong thing: I don't know if there is a naming scheme that would make this clear. `Essentials` seemed appealing because it suggests that you should probably use it. Perhaps documentation is a better approach (i.e. just explain that all packages with Python libraries should inherit PythonExtension, but only those using Python-based installation mechanisms should inherit PythonPackage).